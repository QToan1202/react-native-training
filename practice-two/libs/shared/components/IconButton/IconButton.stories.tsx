import { ComponentMeta, ComponentStory } from '@storybook/react-native'
import { action } from '@storybook/addon-actions'

import { IconButton } from '../index'

export default {
  title: 'components/IconButton',
  component: IconButton,
} as ComponentMeta<typeof IconButton>

const Template: ComponentStory<typeof IconButton> = (args) => <IconButton {...args} />

export const Default = Template.bind({})
Default.args = {
  icon: require('@assets/favicon.png'),
  accessibilityLabel: 'like button',
  onPress: action('Pressing'),
}

export const ButtonWithoutBg = Template.bind({})
ButtonWithoutBg.args = {
  icon: require('@assets/favicon.png'),
  noBackground: true,
  onPress: action('Pressing'),
}
