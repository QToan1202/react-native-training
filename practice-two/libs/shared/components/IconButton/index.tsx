import { memo } from 'react'
import isEqual from 'react-fast-compare'

import StyledIconButton, { CustomButtonProps } from './styles'

export type IconButtonProps = CustomButtonProps

const IconButton = ({ children, ...rest }: IconButtonProps) => {
  return <StyledIconButton {...rest}>{children}</StyledIconButton>
}

export default memo(IconButton, isEqual)
