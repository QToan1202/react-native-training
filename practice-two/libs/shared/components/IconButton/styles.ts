import { Circle, GetProps, styled } from 'tamagui'

const CustomButton = styled(Circle, {
  name: 'IconButton',
  alignSelf: 'flex-start',
  padding: '$space.2',
  backgroundColor: '$color.blur_white',
  pressStyle: {
    opacity: 0.6,
  },

  variants: {
    noBackground: {
      true: {
        backgroundColor: '$color.transparent',
      },
    },
  } as const,
  defaultVariants: {
    noBackground: false,
  },
})

export type CustomButtonProps = GetProps<typeof CustomButton>

export default CustomButton
