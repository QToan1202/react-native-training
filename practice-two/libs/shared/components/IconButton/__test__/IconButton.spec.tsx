import { render, fireEvent } from '@testing-library/react-native'

import IconButton from '../index'

describe('Testing icon button component', () => {
  it('Render button component with an icon', () => {
    const mockOnPress = jest.fn()
    const component = render(
      <IconButton icon={require('@assets/favicon.png')} onPress={mockOnPress} noBackground />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('Press on button with fire onPress event', () => {
    const mockOnPress = jest.fn()
    mockOnPress.mockReturnValue(true)

    const { getByLabelText } = render(
      <IconButton
        icon={require('@assets/favicon.png')}
        onPress={mockOnPress}
        accessibilityLabel="icon button"
      />
    )

    fireEvent.press(getByLabelText('icon button'))
    expect(mockOnPress).toBeCalledTimes(1)
    expect(mockOnPress.mock.results[0].value).toBeTruthy()
  })
})
