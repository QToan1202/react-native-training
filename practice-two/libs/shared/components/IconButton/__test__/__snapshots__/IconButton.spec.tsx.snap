// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing icon button component Render button component with an icon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderRadius": 50,
      "opacity": 1,
      "padding": 7,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/favicon.png",
      }
    }
    style={
      {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
